jerry-debugger/jerry-client-ws.html

```diff
@@ -0,0 +1,970 @@
+    breakpoint.activeIndex = -1;
+
+    var values = [ JERRY_DEBUGGER_UPDATE_BREAKPOINT,
+                   0,
+                   breakpoint.func.byte_code_cp,
+                   breakpoint.offset ];
+
+    encodeMessage("BBCI", values);
+
+    appendLog("Breakpoint " + index + " is deleted.");
+  }
+
+  this.listBreakpoints = function()
+  {
+    appendLog("List of active breakpoints:");
+    var found = false;
+
+    for (var i in activeBreakpoints)
+    {
+      appendLog("  breakpoint " + i + " at " + breakpointToString(activeBreakpoints[i]));
+      found = true;
+    }
+
+    if (!found)
+    {
+      appendLog("  no active breakpoints");
+    }
+  }
+
+  this.dump = function()
+  {
+    for (var i in functions)
+    {
+      var func = functions[i];
+      var resource = func.resource;
+
+      if (resource == '')
+      {
+        resource = "<unknown>";
+      }
+
+      appendLog("Function 0x" + Number(i).toString(16) + " '" + func.name + "' at " + resource + ":" + func.firstLine);
+
+      for (var j in func.lines)
+      {
+        var active = "";
+
+        if (func.lines[j].active >= 0)
+        {
+          active = " (active: " + func.lines[j].active + ")";
+        }
+
+        appendLog("  Breatpoint line: " + j + " at memory offset: " + func.lines[j].offset + active);
+      }
+    }
+  }
+}
+
+function debuggerCommand(event)
+{
+  if (event.keyCode != 13)
+  {
+    return true;
+  }
+
+  var command = commandBox.value.trim();
+
+  args = /^([a-zA-Z]+)(?:\s+([^\s].*)|)$/.exec(command);
+
+  if (!args)
+  {
+    appendLog("Invalid command");
+    document.getElementById("command").value = "";
+    return true;
+  }
+
+  if (!args[2])
+  {
+    args[2] = "";
+  }
+
+  if (args[1] == "help")
+  {
+    appendLog("Debugger commands:\n" +
+              "  connect IP - connect to server\n" +
+              "  break|b file_name:line|function_name - set breakpoint\n" +
+              "  delete|d id - delete breakpoint\n" +
+              "  list - list breakpoints\n" +
+              "  continue|c - continue execution\n" +
+              "  step|s - step-in execution\n" +
+              "  next|n - connect to server\n" +
+              "  backtrace|bt [max-depth] - get backtrace\n" +
+              "  dump - dump all breakpoint data");
+
+    commandBox.value = "";
+    return true;
+  }
+
+  if (args[1] == "connect")
+  {
+    if (debuggerObj)
+    {
+      appendLog("Debugger is connected");
+      return true;
+    }
+
+    if (args[2] == "")
+    {
+      appendLog("IP address expected");
+      return true;
+    }
+
+    appendLog("Connect to: " + args[2]);
+
+    debuggerObj = new DebuggerClient(args[2]);
+
+    commandBox.value = "";
+    return true;
+  }
+
+  if (!debuggerObj)
+  {
+    appendLog("Debugger is NOT connected");
+
+    commandBox.value = "";
+    return true;
+  }
+
+  switch(args[1])
+  {
+    case "b":
+    case "break":
+      debuggerObj.setBreakpoint(args[2]);
+
+      commandBox.value = "";
```

Comment:
As the `commandBox.value` and `return true` is almost in every case we could simple do a break here and add the required lines at the end of the function imho. Or maybe even better that after we read the command value simply clear it out there.